File: AngularFrontend/src/app/models/taskgroup.ts
import { ITaskGroup } from '../models/taskgroup.interface';
import { ItemType } from '../utils/enum.types';
import { ITaskItem } from './taskitem.interface';
export class TaskGroup implements ITaskGroup {
  id?: number;
  type: ItemType = ItemType.Family;
  taskItems: ITaskItem[]=[];
}

File: WebAPI/Models/TaskGroup.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WebAPI.Utlis;

namespace WebAPI.Models
{

    public class TaskGroup
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        public ItemType Type { get; set; }

        // Navigation properties
        public virtual ICollection<TaskItem> TaskItems { get; set; } = new List<TaskItem>();
       

    }
}
